R/model_fairness.R

```diff
@@ -238,181 +241,185 @@ model_fairness <- function(data,
-  colnames(odds.ratio[["all"]]) <- if (is_bayesian) c("estimate", "lb", "ub", "bf10") else c("estimate", "lb", "ub", "p.value")
-  for (i in seq_len(nlevels(data[, protected]))) {
-    group <- levels(data[, protected])[i]
-    groupDat <- data[data[, protected] == group, ]
-    # Confusion matrices for each group
-    confmat[[group]][["matrix"]] <- table("Actual" = groupDat[, target], "Predicted" = groupDat[, predictions])
-    confmat[[group]][["tp"]] <- tp <- confmat[[group]][["matrix"]][positive, positive]
-    confmat[[group]][["fp"]] <- fp <- sum(confmat[[group]][["matrix"]][negative, positive])
-    confmat[[group]][["tn"]] <- tn <- sum(confmat[[group]][["matrix"]][negative, negative])
-    confmat[[group]][["fn"]] <- fn <- sum(confmat[[group]][["matrix"]][positive, negative])
-    confmat[[group]][["n"]] <- sum(confmat[[group]][["matrix"]])
-    # Performance measures for each group
-    performance[[group]][["support"]] <- performance[["all"]][i, 1] <- sum(confmat[[group]][["matrix"]])
-    performance[[group]][["accuracy"]] <- performance[["all"]][i, 2] <- (confmat[[group]][["tp"]] + confmat[[group]][["tn"]]) / (confmat[[group]][["tp"]] + confmat[[group]][["tn"]] + confmat[[group]][["fp"]] + confmat[[group]][["fn"]])
-    performance[[group]][["precision"]] <- performance[["all"]][i, 3] <- confmat[[group]][["tp"]] / (confmat[[group]][["tp"]] + confmat[[group]][["fp"]])
-    performance[[group]][["recall"]] <- performance[["all"]][i, 4] <- confmat[[group]][["tp"]] / (confmat[[group]][["tp"]] + confmat[[group]][["fn"]])
-    performance[[group]][["f1.score"]] <- performance[["all"]][i, 5] <- 2 * ((performance[[group]][["precision"]] * performance[[group]][["recall"]]) / (performance[[group]][["precision"]] + performance[[group]][["recall"]]))
-    if (metric == "dp") {
-      metrics[[group]][["estimate"]] <- metrics[["all"]][i, 1] <- tp + fp
-    } else {
-      metrics[[group]][["numerator"]] <- switch(metric,
-        "pp" = tp + fp,
-        "prp" = tp,
-        "ap" = tp + tn,
-        "fnrp" = fn,
-        "fprp" = fp,
-        "tprp" = tp,
-        "npvp" = tn,
-        "sp" = tn
-      )
-      metrics[[group]][["denominator"]] <- switch(metric,
-        "pp" = tp + fp + tn + fn,
-        "prp" = tp + fp,
-        "ap" = tp + fp + tn + fn,
-        "fnrp" = tp + fn,
-        "fprp" = tn + fp,
-        "tprp" = tp + fn,
-        "npvp" = tn + fn,
-        "sp" = tn + fp
-      )
-    }
-  }
-  names(confmat) <- groups
-  # Sample estimates for each group
-  if (metric != "dp") {
+  if (metric == "eo") {
+    stop("Error: 'Equalized Odds' (EO) requires simultaneous evaluation of 'True Positive Rate' (TPR) and 'False Positive Rate' (FPR). You must first select tprp, then fprp, and ensure both fairness measures conclude no discrimination for both privileged and unprivileged groups before concluding no discrimination according to EO.")
+  } else {
```

Comment:
Remove the `else` here, you quit the analysis at this point anyway if metric = "eo"